test/performance/broker-latency/main.go

```diff
@@ -0,0 +1,286 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/transport/http"
+	"github.com/golang/protobuf/proto"
+	"github.com/google/mako/helpers/go/quickstore"
+	qpb "github.com/google/mako/helpers/proto/quickstore/quickstore_go_proto"
+	"golang.org/x/sync/errgroup"
+	"knative.dev/eventing/test/performance"
+)
+
+const (
+	defaultEventType   = "perf-test-event-type"
+	defaultEventSource = "perf-test-event-source"
+
+	// The interval and timeout used for polling in checking event states.
+	pollInterval = 1 * time.Second
+	pollTimeout  = 4 * time.Minute
+)
+
+// flags for the image
+var (
+	benchmark          string
+	sinkURL            string
+	msgSize            int
+	eventNum           int
+	errorRateThreshold float64
+	timeout            int
+	encoding           string
+	eventTimeMap       map[string]chan time.Time
+	resultCh           chan state
+)
+
+// eventStatus is status of the event, for now only if all events are in received status can the
+// test be considered as PASS.
+type eventStatus int
+
+const (
+	sent eventStatus = iota
+	received
+	undelivered
+	dropped
+	duplicated
+	corrupted // TODO(Fredy-Z): corrupted status is not being used now
+)
+
+// state saves the data that is used to generate the metrics
+type state struct {
+	latency time.Duration
+	status  eventStatus
+}
+
+const (
+	makoAddr = "localhost:9813"
+)
+
+func init() {
+	flag.StringVar(&benchmark, "benchmark", "", "The mako benchmark ID")
```

Comment:
You should pick up the `mako.Setup()` stuff from pkg (cc @chaodaiG ), you'll start to get a lot of interesting auto-tagging (e.g. K8s version, commit, project-id) and will handle some environmental stuff like the benchmark key for you, so folks don't have to manually context switch when running things on their own cluster.